ui/components/component-library/base-icon/base-icon.stories.js

```diff
@@ -0,0 +1,180 @@
+    <BaseIcon {...args} size={SIZES.SM} />
+    <BaseIcon {...args} size={SIZES.MD} />
+    <BaseIcon {...args} size={SIZES.LG} />
+    <BaseIcon {...args} size={SIZES.XL} />
+  </Box>
+);
+
+export const Color = (args) => (
+  <Box display={DISPLAY.FLEX} alignItems={ALIGN_ITEMS.BASELINE}>
+    <Box padding={1} display={DISPLAY.FLEX} alignItems={ALIGN_ITEMS.CENTER}>
+      <BaseIcon {...args} color={COLORS.INHERIT} />
+    </Box>
+    <Box padding={1} display={DISPLAY.FLEX} alignItems={ALIGN_ITEMS.CENTER}>
+      <BaseIcon {...args} color={COLORS.ICON_DEFAULT} />
+    </Box>
+    <Box padding={1} display={DISPLAY.FLEX} alignItems={ALIGN_ITEMS.CENTER}>
+      <BaseIcon {...args} color={COLORS.ICON_ALTERNATIVE} />
+    </Box>
+    <Box padding={1} display={DISPLAY.FLEX} alignItems={ALIGN_ITEMS.CENTER}>
+      <BaseIcon {...args} color={COLORS.ICON_MUTED} />
+    </Box>
+    <Box
+      padding={1}
+      display={DISPLAY.FLEX}
+      alignItems={ALIGN_ITEMS.CENTER}
+      backgroundColor={COLORS.OVERLAY_DEFAULT}
+    >
+      <BaseIcon {...args} color={COLORS.OVERLAY_INVERSE} />
+    </Box>
+    <Box padding={1} display={DISPLAY.FLEX} alignItems={ALIGN_ITEMS.CENTER}>
+      <BaseIcon {...args} color={COLORS.PRIMARY_DEFAULT} />
+    </Box>
+    <Box
+      padding={1}
+      display={DISPLAY.FLEX}
+      alignItems={ALIGN_ITEMS.CENTER}
+      backgroundColor={COLORS.PRIMARY_DEFAULT}
+    >
+      <BaseIcon {...args} color={COLORS.PRIMARY_INVERSE} />
+    </Box>
+    <Box padding={1} display={DISPLAY.FLEX} alignItems={ALIGN_ITEMS.CENTER}>
+      <BaseIcon {...args} color={COLORS.ERROR_DEFAULT} />
+    </Box>
+    <Box
+      padding={1}
+      display={DISPLAY.FLEX}
+      alignItems={ALIGN_ITEMS.CENTER}
+      backgroundColor={COLORS.ERROR_DEFAULT}
+    >
+      <BaseIcon {...args} color={COLORS.ERROR_INVERSE} />
+    </Box>
+    <Box padding={1} display={DISPLAY.FLEX} alignItems={ALIGN_ITEMS.CENTER}>
+      <BaseIcon {...args} color={COLORS.SUCCESS_DEFAULT} />
+    </Box>
+    <Box
+      padding={1}
+      display={DISPLAY.FLEX}
+      alignItems={ALIGN_ITEMS.CENTER}
+      backgroundColor={COLORS.SUCCESS_DEFAULT}
+    >
+      <BaseIcon {...args} color={COLORS.SUCCESS_INVERSE} />
+    </Box>
+    <Box padding={1} display={DISPLAY.FLEX} alignItems={ALIGN_ITEMS.CENTER}>
+      <BaseIcon {...args} color={COLORS.WARNING_DEFAULT} />
+    </Box>
+    <Box
+      padding={1}
+      display={DISPLAY.FLEX}
+      alignItems={ALIGN_ITEMS.CENTER}
+      backgroundColor={COLORS.WARNING_DEFAULT}
+    >
+      <BaseIcon {...args} color={COLORS.WARNING_INVERSE} />
+    </Box>
+    <Box padding={1} display={DISPLAY.FLEX} alignItems={ALIGN_ITEMS.CENTER}>
+      <BaseIcon {...args} color={COLORS.INFO_DEFAULT} />
+    </Box>
+    <Box
+      padding={1}
+      display={DISPLAY.FLEX}
+      alignItems={ALIGN_ITEMS.CENTER}
+      backgroundColor={COLORS.INFO_DEFAULT}
+    >
+      <BaseIcon {...args} color={COLORS.INFO_INVERSE} />
+    </Box>
+  </Box>
+);
```

Comment:
I updated this to map `ICON_COLORS` but I think this could be used as a good resource for engineers to copy and paste if they require a certain color combination? But can change it back if it doesn't seem like a used resource

```jsx
<Box display={DISPLAY.FLEX} alignItems={ALIGN_ITEMS.BASELINE}>
    {Object.keys(ICON_COLORS).map((color) => {
      const isInverseColor = color.indexOf('_INVERSE') > -1;
      const defaultColorOfInverse = `${color.replace(/_INVERSE/g, '')}_DEFAULT`;
      return (
        <Box
          key={color}
          padding={1}
          display={DISPLAY.FLEX}
          alignItems={ALIGN_ITEMS.CENTER}
          backgroundColor={
            isInverseColor ? COLORS[defaultColorOfInverse] : null
          }
        >
          <BaseIcon {...args} color={COLORS[color]} />
        </Box>
      );
    })}
  </Box>
```